-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-35000/24856

ERROR: Build errors found during clang-tidy run.
```
RecoPixelVertexing/PixelLowPtUtilities/src/ClusterShapeTrajectoryFilter.cc:45:22: error: no member named 'getData' in 'edm::Event' [clang-diagnostic-error]
  theCache = &iEvent.getData(theCacheToken);
                     ^
Suppressed 1103 warnings (1102 in non-user code, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/RecoPixelVertexing/PixelLowPtUtilities/interface/ClusterShapeTrajectoryFilter.h
+++ b/RecoPixelVertexing/PixelLowPtUtilities/interface/ClusterShapeTrajectoryFilter.h
@@ -36,6 +36,7 @@ public:
 
 private:
   edm::EDGetTokenT<SiPixelClusterShapeCache> theCacheToken;
+  edm::ESGetToken<ClusterShapeHitFilter, TrajectoryFilter::Record> theFilterToken;
   const SiPixelClusterShapeCache* theCache;
   const ClusterShapeHitFilter* theFilter;
 };

--- a/RecoPixelVertexing/PixelLowPtUtilities/src/ClusterShapeTrajectoryFilter.cc
+++ b/RecoPixelVertexing/PixelLowPtUtilities/src/ClusterShapeTrajectoryFilter.cc
@@ -6,14 +6,12 @@
 
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/EventSetup.h"
-#include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "TrackingTools/PatternTools/interface/Trajectory.h"
 #include "TrackingTools/PatternTools/interface/TempTrajectory.h"
 
-#include "DataFormats/Common/interface/Handle.h"
 #include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
 #include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
 #include "DataFormats/TrackerRecHit2D/interface/SiStripRecHit2D.h"
@@ -37,18 +35,14 @@ using namespace std;
 /*****************************************************************************/
 ClusterShapeTrajectoryFilter::ClusterShapeTrajectoryFilter(const edm::ParameterSet& iConfig, edm::ConsumesCollector& iC)
     : theCacheToken(iC.consumes<SiPixelClusterShapeCache>(iConfig.getParameter<edm::InputTag>("cacheSrc"))),
+      theFilterToken(iC.esConsumes(edm::ESInputTag("", "ClusterShapeHitFilter"))),
       theFilter(nullptr) {}
 
 ClusterShapeTrajectoryFilter::~ClusterShapeTrajectoryFilter() {}
 
 void ClusterShapeTrajectoryFilter::setEvent(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
-  edm::ESHandle<ClusterShapeHitFilter> shape;
-  iSetup.get<TrajectoryFilter::Record>().get("ClusterShapeHitFilter", shape);
-  theFilter = shape.product();
-
-  edm::Handle<SiPixelClusterShapeCache> cache;
-  iEvent.getByToken(theCacheToken, cache);
-  theCache = cache.product();
+  theFilter = &iSetup.getData(theFilterToken);
+  theCache = &iEvent.get(theCacheToken);
 }
 
 /*****************************************************************************/

